lang/java/compiler/src/main/java/org/apache/avro/compiler/specific/SpecificCompiler.java

```diff
@@ -287,15 +287,52 @@ public void addCustomConversion(Class<?> conversionClass) {
   }
 
   public Collection<String> getUsedConversionClasses(Schema schema) {
```

Comment:
There's some dead code as a consequence of this change: `getClassNamesOfPrimitiveFields` -- I'm taking a look.

I'm also thinking that if the user explicitly called `addCustomConversion`, it should be considered "used" and included no matter what, but that currently isn't the case.